app/generic/verifier/verifier.py

```diff
@@ -36,6 +39,103 @@ def __get_verifier_by_version(protocol_version: str) -> Verifier:
         raise ValueError(f"Unknown protocol version {protocol_version}")
 
 
+@router.post("/credentials")
+async def get_credentials(
+    proof_request: ProofRequestGeneric,
```

Comment:
`ProofRequestGeneric` has `proof_id` as optional parameter, but it is required for this endpoint. Maybe we can make a separate type that marks the `proof_id` as required?

Reply:
Let me have a look, can't think of it rn from the top of my head, but there was a reason I did it this way... Happy to change it to your suggestion, though.

Reply:
made it required now

Reply:
Hmm. `ProofRequestGeneric` depends on `ProofRequestBase` which has a  `protocol_version`. I don't think that's necessary for this call right, as we can extract it from the `proof_id`?